### PR TITLE
UAG-374 UI/UX - Post(Layout) Blocks

### PR DESCRIPTION
**Pull request by supriyajjadhav**
### Description
 Card - https://ua-gutenberg.atlassian.net/browse/UAG-374

### Types of changes
non-breaking change which adds functionality

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/post/styling.js [hunk 1]
```diff
@@ -102,70 +169,81 @@ function styling( props ) {
...
 		' .uagb-post__items article': {
-			'padding-right': generateCSSUnit( rowGap / 2, 'px' ),
-			'padding-left': generateCSSUnit( rowGap / 2, 'px' ),
-			'margin-bottom': generateCSSUnit( columnGap, 'px' ),
+			'padding-right': generateCSSUnit( rowGap / 2, rowGapUnit ),
+			'padding-left': generateCSSUnit( rowGap / 2, rowGapUnit ),
+			'margin-bottom': generateCSSUnit( columnGap, columnGapUnit  ),
 		},
 		' .uagb-post__inner-wrap': {
 			background: bgColor,
```
Comment (AbhijitNage123) [inline src/blocks/post/styling.js:209]: Single quotes missing for some of the styles. Please add a single quote in styling.js

File: src/blocks/post/post-grid/settings.js [hunk 2]
```diff
@@ -2,236 +2,377 @@
...
+		 displayPostLink,
+		 newTab,
+		 ctaText,
+		 borderWidth,
+		 borderStyle,
+		 borderColor,
+		 borderHColor,
+		 borderRadius,
+		 btnVPadding,
+		 btnHPadding,
```
Comment (AbhijitNage123) [inline src/blocks/post/post-grid/settings.js:68]: Remove this if not using btnVPadding, btnHPadding.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -3331,7 +3368,11 @@ public static function get_post_selectors( $attr ) {
 				);
 				$selectors[' .uagb-post__text .uagb-post__cta a']                           = array(
 					'color'   => $attr['ctaColor'],
-					'padding' => ( $attr['btnVPadding'] ) . 'px ' . ( $attr['btnHPadding'] ) . 'px',
+					'padding-top' => UAGB_Helper::get_css_value( $attr['paddingBtnTop'] , $attr['paddingBtnUnit'] ),
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:3371]: Need to maintain backward compatibility here for horizontal and vertical padding option

File: classes/class-uagb-block-helper.php [hunk 4]
```diff
@@ -3303,9 +3340,9 @@ public static function get_post_selectors( $attr ) {
 			);
 			if ( 'background' !== $attr['imgPosition'] ) {
 				$selectors[' .uagb-post__inner-wrap > .uagb-post__image:first-child'] = array(
-					'margin-top'   => UAGB_Helper::get_css_value( ( -$attr['contentPadding'] ), 'px' ),
-					'margin-left'  => UAGB_Helper::get_css_value( ( -$attr['contentPadding'] ), 'px' ),
-					'margin-right' => UAGB_Helper::get_css_value( ( -$attr['contentPadding'] ), 'px' ),
+					'margin-top'   => UAGB_Helper::get_css_value( ( -$attr['paddingTop'] ), $attr['contentPaddingUnit'] ),
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:3343]: We have a change attribute `contentPadding` to `paddingTop` we need to check backward compatibility here too.

File: classes/class-uagb-block-helper.php [hunk 5]
```diff
@@ -3277,24 +3311,27 @@ public static function get_post_selectors( $attr ) {
...
-					'margin-bottom' => UAGB_Helper::get_css_value( $attr['titleBottomSpace'], 'px' ),
+					'margin-bottom' => UAGB_Helper::get_css_value( $attr['titleBottomSpace'], $attr['titleBottomSpaceUnit'] ),
 				),
 				' .uagb-post-grid-byline'    => array(
-					'margin-bottom' => UAGB_Helper::get_css_value( $attr['metaBottomSpace'], 'px' ),
+					'margin-bottom' => UAGB_Helper::get_css_value( $attr['metaBottomSpace'], $attr['metaBottomSpaceUnit'] ),
 				),
 				' .uagb-post__excerpt'       => array(
-					'margin-bottom' => UAGB_Helper::get_css_value( $attr['excerptBottomSpace'], 'px' ),
+					'margin-bottom' => UAGB_Helper::get_css_value( $attr['excerptBottomSpace'], $attr['excerptBottomSpaceUnit'] ),
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:3334]: Always assign default value to unit. excerptBottomSpaceUnit to 'px'

File: classes/class-uagb-block-helper.php [hunk 6]
```diff
@@ -3357,7 +3398,17 @@ public static function get_post_mobile_selectors( $attr ) {
 
 			return array(
 				' .uagb-post__inner-wrap' => array(
-					'padding' => ( $attr['contentPaddingMobile'] ) . 'px',
+					'padding-top' => UAGB_Helper::get_css_value( $attr[' paddingTopMobile'], $attr['mobilePaddingUnit'] ),
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:3401]: Same here. Need to check backward compatibility and default value to unit to px

**Review by AbhijitNage123 — COMMENTED** [review]
- If we are using a new attribute instead old then we need to maintain backward compatibility for the same.
- Always assign a default value to unit
- Single quote missing for some of style in styling.js

**Review by RajkiranBagal — APPROVED** [review]

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**